### DISCUSSION TITLE
Rewrite core to TypeScript

### DISCUSSION THREAD

**Discussion by robertpiosik** [strapi/strapi#6059] [Legacy Topics]
Hey 👋I was wondering what is the most recent status on TypeScript in terms of Strapi development? Why is the core being still written in JS? It's always impressive to see such big projects maintained without static typing but... 

...TypeScript really helps. And we could shift incrementally.

**Comment by lauriejim** [comment]
Hello!
Here is the item in the roadmap - https://portal.productboard.com/strapi/1-public-roadmap/c/30-support-typescript
And a long issue thread about this topic - https://github.com/strapi/strapi/issues/630

**Reply by robertpiosik** [reply]
Yeah, I've found it later on. Sorry for not closing the topic.
